Documentation/seccomp-guide.md

```diff
@@ -0,0 +1,375 @@
+ * `@systemd/mount` for filesystem mounting and unmounting
+ * `@systemd/network-io` for socket I/O operationgs
+ * `@systemd/obsolete` for unusual, obsolete or unimplemented syscalls
+ * `@systemd/privileged` for syscalls which need super-user syscalls
+ * `@systemd/process` for syscalls acting on process control, execution and namespacing
+ * `@systemd/raw-io` for raw I/O port access
+
+When no seccomp filtering is specified, by default rkt whitelists all the generic
+syscalls typically needed by applications for common operations. This is
+the same set defined by `@rkt/default-whitelist`.
+
+The default set is tailored to stop applications from performing a large
+variety of privileged actions, while not impacting their normal behavior.
+Operations which are typically not needed in containers and which may
+impact host state, eg. invoking `umount(2)`, are denied in this way.
+
+However, this default set is mostly meant as a safety precaution against erratic
+and misbehaving applications, and will not suffice against tailored attacks.
+As such, it is recommended to fine-tune seccomp filtering using one of the
+customizable isolators available in rkt.
+
+## Seccomp Isolators
+
+When running Linux containers, rkt provides two mutually exclusive isolators
+to define a seccomp filter for an application:
+
+* `os/linux/seccomp-retain-set`
+* `os/linux/seccomp-remove-set`
+
+Those isolators cover different use-cases and employ different techniques to
+achieve the same goal of limiting available syscalls. As such, they cannot
+be used together at the same time, and recommended usage varies on a
+case-by-case basis.
+
+### Operation mode
+
+Seccomp isolators work by defining a set of syscall than can be either blocked
+("remove-set") or allowed ("retain-set"). Once an application tries to invoke
+a blocked syscall, the kernel will deny this operation and the application will
+be notified about the failure.
+
+By default, invoking blocked syscalls will result in the application being
+immediately terminated with a `SIGSYS` signal. This behavior can be tweaked by
+returning a specific error code ("errno") to the application instead of
+terminating it.
+
+For both isolators, this can be customized by specifying an additional `errno`
+paramater with the desired symbolic errno name. For a list of errno labels, see
+`man 3 errno`.
+
+### Retain-set
+
+`os/linux/seccomp-retain-set` allows for an additive approach to build a seccomp
+filter: applications will not able to use any syscalls, except the ones
+listed in this isolator.
+
+This whitelisting approach is useful for completely locking down environments
+and whenever application requirements (in terms of syscalls) are
+well-defined in advance. It allows one to ensure that exactly and only the
+specified syscalls could ever be used.
+
+For example, an application that will only need to perform socket I/O operations,
+may have `@systemd/network-io` as the only entry in its "retain-set".
```

Comment:
You raised a good point. I guess the example is a bit too strict due to the "only" wording. I guess it could be turned into a better example by saying that a typical network application will require generic POSIX operations (@systemd/default-whitelist), socket operations (@systemd/network-io) and reacting to I/O events (@systemd/io-event). Seems better?